src/dmd/dinterpret.d

```diff
@@ -698,6 +698,17 @@ private Expression interpretFunction(UnionExp* pue, FuncDeclaration fd, InterSta
     return e;
 }
 
+void incUsageCtfe(InterState* istate, const ref Loc loc)
+{
+    if (global.params.cov && istate)
```

Comment:
Note the test for istate. In nearly every call to this function, istate is guaranteed to be not null. Given that this is going to make CTFE slow, eliminating the redundant check seems worthwhile.

Reply:
I'd also move the cov switch into istate so this call become much more efficient.

Reply:
It is not clear that it would become more efficient at least not on modern cpus.
`jnz global.params.cov` is a perfectly predictable branch, the value cannot ever change during the execution of dmd.
Therefore, it doesn't cost you anything, istate on the other hand can in some cases, be null.

I am going to test this.

Reply:
Results: 
  './dmd_ctfecov_first -cov -ctfe-cov -version=ctfe rx.d -c' ran
    1.01 ± 0.02 times faster than './dmd_istate_frist -cov -ctfe-cov -version=ctfe rx.d -c'
And also:
  './dmd_istate_frist -cov -ctfe-cov -version=ctfe rx.d -c' ran
    1.01 ± 0.02 times faster than './dmd_ctfecov_first -cov -ctfe-cov -version=ctfe rx.d -c'

The difference between the tests was the order in which I ran the binaries.
Which supports my assumption that it doesn't actually matter.